File: app/src/main/java/com/example/rain/net/retrofit/RetrofitS.java
package com.example.rain.net.retrofit;


import com.example.rain.net.call.StringRequest;

import java.util.Map;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.scalars.ScalarsConverterFactory;

public class RetrofitS extends BaseRetrofit {

    private static StringRequest call;

    static  {
        retrofit = new Retrofit.Builder()
                .baseUrl("https://os.zhijierongxing.com")
                .client(initClient())
                .addConverterFactory(ScalarsConverterFactory.create())
                .build();
        call = retrofit.create(StringRequest.class);
    }


    public static Call<String> GET(String path){
        return call.GET(path);
    }

    public static Call<String> GET(String path, Map<String, String> from){
        return call.GET(path,from);
    }

    public static Call<String> GET_PART(String path, Map<String, RequestBody> params){
        return call.GET_PART(path,params);
    }

    public static Call<String> POST(String path){
        return call.POST(path);
    }

    public static Call<String> POST(String path, Map<String, String> from){
        return call.POST(path,from);
    }


    public static  Call<String> POST_BODY( String path, Map<String, String> body){
        return call.POST_BODY(path,body);
    }

    public static Call<String> POST_PART(String path, Map<String, RequestBody> params){
        return call.POST_PART(path,params);
    }


    public static Call<String> POST(String path, Map<String, String> form, Map<String, String> body){
        return  call.POST(path,form,body);
    }


    public static Call<String> DELETE( String path, Map<String, RequestBody> params){
        return  call.DELETE(path,params);
    }

    public static Call<String> PUT( String path, Map<String, RequestBody> params){
        return  call.PUT(path,params);
    }

    // 上传
    public static Call<ResponseBody> UPLOAD(String path, Map<String, RequestBody> params){
        return call.UPLOAD(path,params); }

    // 下载
    public static  Call<ResponseBody> DOWNLOAD(String path){
        return call.DOWNLOAD(path);
    }

}

File: app/src/main/java/com/example/rain/net/retrofit/RetrofitS.kt
package com.example.rain.net.retrofit

import com.example.rain.net.call.StringRequest
import okhttp3.RequestBody
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.scalars.ScalarsConverterFactory

class RetrofitS :BaseRetrofit {

    private var call: StringRequest? = null

    constructor(){
        retrofit = Retrofit.Builder()
            .baseUrl(baseUrl)
            .client(initClient())
            .addConverterFactory(ScalarsConverterFactory.create())
            .build()
        call = retrofit?.create(StringRequest::class.java)
    }

    fun GET(path: String?): Call<String?>? {
        return call?.GET(path)
    }

    fun GET(path: String?, from: Map<String?, String?>?): Call<String?>? {
        return call?.GET(path, from)
    }

    fun GET_PART(path: String?, params: Map<String?, RequestBody?>?): Call<String?>? {
        return call?.GET_PART(path, params)
    }

    fun POST(path: String?): Call<String?>? {
        return call?.POST(path)
    }

    fun POST(path: String?, from: Map<String?, String?>?): Call<String?>? {
        return call?.POST(path, from)
    }


    fun POST_BODY(path: String?, body: Map<String?, String?>?): Call<String?>? {
        return call?.POST_BODY(path, body)
    }

    fun POST_PART(path: String?, params: Map<String?, RequestBody?>?): Call<String?>? {
        return call?.POST_PART(path, params)
    }


    fun POST(
        path: String?,
        form: Map<String?, String?>?,
        body: Map<String?, String?>?
    ): Call<String?>? {
        return call?.POST(path, form, body)
    }


    fun DELETE(path: String?, params: Map<String?, RequestBody?>?): Call<String?>? {
        return call?.DELETE(path, params)
    }

    fun PUT(path: String?, params: Map<String?, RequestBody?>?): Call<String?>? {
        return call?.PUT(path, params)
    }

    // 上传
    fun UPLOAD(path: String?, params: Map<String?, RequestBody?>?): Call<ResponseBody?>? {
        return call?.UPLOAD(path, params)
    }

    // 下载
    fun DOWNLOAD(path: String?): Call<ResponseBody?>? {
        return call?.DOWNLOAD(path)
    }

}